modules/abstract-utxo/src/abstractUtxoCoin.ts

```diff
@@ -695,10 +696,13 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
     // verify the signature against the user public key
     assert(userKeychain.pub);
     const publicKey = bip32.fromBase58(userKeychain.pub).publicKey;
+    // Due to interface of `bitcoinMessage`, we need to convert the public key to an address.
+    // Note that this address has no relationship to on-chain transactions. We are
+    // only interested in the address as a representation of the public key.
     const signingAddress = utxolib.address.toBase58Check(
       utxolib.crypto.hash160(publicKey),
       utxolib.networks.bitcoin.pubKeyHash,
-      this.network
+      utxolib.networks.bitcoin
```

Comment:
```suggestion
      // we do not pass `this.network` here because it would fail for zcash
      // the bitcoinMessage library decodes the address and throws away the first byte
      // because zcash has a two-byte prefix, verify() decodes zcash addresses to an invalid pubkey hash
      utxolib.networks.bitcoin
```